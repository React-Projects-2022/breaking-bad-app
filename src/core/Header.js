import { NavLink } from "react-router-dom";
import { PATHS } from "./../helpers/paths";
import { useTranslation } from "react-i18next";
const Header = () => {
  
const { t } = useTranslation();
  return (
    <nav className="navbar navbar-expand-lg navbar-dark bg-dark sticky-top">
      <div className="container-fluid">
        <button
          className="navbar-toggler"
          type="button"
          data-bs-toggle="collapse"
          data-bs-target="#navbarTogglerDemo01"
          aria-controls="navbarTogglerDemo01"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <span className="navbar-toggler-icon"></span>
        </button>
        <div className="collapse navbar-collapse" id="navbarTogglerDemo01">
          <NavLink to={PATHS.HOME} className="navbar-brand">
            Breaking Bad
          </NavLink>
          <ul className="navbar-nav me-auto mb-2 mb-lg-0">
            <li className="nav-item">
              <NavLink
                to={PATHS.HOME}
                className={({ isActive }) =>
                  isActive ? "nav-link active" : "nav-link"
                }
              >
                { t("navbar.home") }
              </NavLink>
            </li>
            <li className="nav-item">
              <NavLink
                to={PATHS.ALL_CHARACTERS}
                className={({ isActive }) =>
                  isActive ? "nav-link active" : "nav-link"
                }
              >
                { t("navbar.characters") }
              </NavLink>
            </li>
            <li className="nav-item">
              <NavLink
                to={PATHS.EPISODES}
                className={({ isActive }) =>
                  isActive ? "nav-link active" : "nav-link"
                }
              >
                { t("navbar.episodes") }
              </NavLink>
            </li>

            <li className="nav-item">
              <NavLink
                to={PATHS.VOTES}
                className={({ isActive }) =>
                  isActive ? "nav-link active" : "nav-link"
                }
              >
                { t("navbar.home") }
              </NavLink>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  );
};

export default Header;
