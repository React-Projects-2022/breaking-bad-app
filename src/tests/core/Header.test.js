import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import Header from "../../core/Header";

describe("Pruebas en Header", () => {
  const wrapper = shallow(<Header />);

  test("Debe de mostrar <Header /> correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de comprobar que tenemos 4 'NavLinks'", () => {
    expect(wrapper.find("NavLink").length).toBe(4);
  });

  test("Comprobando los textos de los 'NavLink'", () => {
    const labels = ["Breaking Bad", "Home", "Episodes", "Votes"];
    labels.map((element, index) =>
      expect(wrapper.find("NavLink").at(index).text()).toBe(element)
    );
  });

  test("Comprobar que se aplica correctamente el estilo implementado", () => {
    const expectClassName =
      "navbar navbar-expand-lg navbar-dark bg-dark sticky-top";
    expect(wrapper.find("nav").props().className).toBe(expectClassName);
  });

  test("Comprobar si los elementos que dan acceso a las pÃ¡ginas son correctos", () => {
    const expectUlClassName = "navbar-nav me-auto mb-2 mb-lg-0";
    expect(wrapper.find("ul").props().className).toBe(expectUlClassName);
    const expectLiClassName = "nav-item";
    expect(wrapper.find("li").at(0).props().className).toBe(expectLiClassName);
    expect(wrapper.find("li").at(1).props().className).toBe(expectLiClassName);
    expect(wrapper.find("li").at(2).props().className).toBe(expectLiClassName);
  });
});
