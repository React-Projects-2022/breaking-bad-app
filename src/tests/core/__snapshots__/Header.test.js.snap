// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Header Debe de mostrar <Header /> correctamente 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark bg-dark sticky-top"
>
  <div
    className="container-fluid"
  >
    <button
      aria-controls="navbarTogglerDemo01"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-bs-target="#navbarTogglerDemo01"
      data-bs-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarTogglerDemo01"
    >
      <NavLink
        className="navbar-brand"
        to="/breaking-bad-app"
      >
        Breaking Bad
      </NavLink>
      <ul
        className="navbar-nav me-auto mb-2 mb-lg-0"
      >
        <li
          className="nav-item"
        >
          <NavLink
            className={[Function]}
            to="/breaking-bad-app"
          >
            Home
          </NavLink>
        </li>
        <li
          className="nav-item"
        >
          <NavLink
            className={[Function]}
            to="/breaking-bad-app/episodes"
          >
            Episodes
          </NavLink>
        </li>
        <li
          className="nav-item"
        >
          <NavLink
            className={[Function]}
            to="/breaking-bad-app/votes"
          >
            Votes
          </NavLink>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
